src/smc-webapp/project/ask-filename.tsx

```diff
@@ -0,0 +1,106 @@
+import { Component, React, Rendered } from "../app-framework";
```

Comment:
I'm moving to importing `React` from "react" using React.Component. Also the `Rendered` type is usually unnecessary since TS can infer it.

Reply:
> Also the Rendered type is usually unnecessary since TS can infer it.

For me the point of putting types in explicitly is it ensures that I don't screw up the code. I explicitly put return types for my functions because it ensures that I write them properly (without implicitly relying on how they are used), and also makes my intention very clear to readers.   I find this extremely useful.  It immediately ensures that all return paths return a rendered component (or undefined), etc.   I am going to use Rendered explicitly in all my code, unless you can give me a really good argument to not do so.   I consider this similar to using strict null checks, which Typescript of course does not have on by default, but prevents a lot of issues if you do it from the beginning.